### ISSUE TITLE
Could You Please Share the Curve of Training Loss?

### ISSUE THREAD

**Issue by semi-supervised-paper** [HobbitLong/CMC#20]
Hi, 
I want to use CMC in my own experiment, but the loss is strange. At each epoch, the loss decays as normal (like from 20 to 11). But at the next epoch, the loss becomes nearly the same as begining (the loss is 20 again). I wonder if it is 'normal' in CMC. 

Thanks.

**Comment by HobbitLong** [comment]
I notice similar pattern, but it only happens for the first few (<5) epochs. Below is an example of the loss:

![ab](https://user-images.githubusercontent.com/29867600/69791232-139a5400-1192-11ea-8f39-9806db209c50.png)
![l](https://user-images.githubusercontent.com/29867600/69791242-19903500-1192-11ea-8b53-608d8b719412.png)

**Comment by semi-supervised-paper** [comment]
Thanks for your kindly response.

**Comment by talshef** [comment]
Hi, thanks for sharing the code.
I've got the same issue, the loss is decay durning the epoch but reversed to the same point in the beginning of the next epoch. did you fin a solution?

**Comment by IFICL** [comment]
I also got the same issue. I guess the reason lead to this situation might be that the features stores in the memory bank come from the previous epoch and leads to high loss when new features come in. But if you average the loss of each epoch, it still decays.

**Comment by ShaoTengLiu** [comment]
I also got the same issue, which has also been discussed in #27. Yes, the average loss still decays, but I find this may hurt the performance of CMC on small datasets.

**Comment by talshef** [comment]
> Yes, the average loss still decays, but I find this may hurt the performance of CMC on small 
I have the same experiment
Tuning the hyperparameters: nce_k, nce_m and learning rate improved the issue in my experiments

**Comment by ShaoTengLiu** [comment]
> > I also got the same issue, which has also been discussed in #27. Yes, the average loss still decays, but I find this may hurt the performance of CMC on small datasets.
> 
> I have the same experiment.
> Tuning the hyperparameters: nce_k, nce_m and learning rate improved the issue in my experiments

Yes, I also find decreasing nce_k can improve the performance. Could you please share some experience on tuning nce_m and lr?

**Comment by talshef** [comment]
decreasing nce_m helped in some case to make the training more stable, meaning that the loss start follow the loss of the last epoch sooner.
For the lr, i played with it until i get the right shape of the loos curve. The same as: 


> ![ab](https://user-images.githubusercontent.com/29867600/69791232-139a5400-1192-11ea-8f39-9806db209c50.png)
> ![l](https://user-images.githubusercontent.com/29867600/69791242-19903500-1192-11ea-8b53-608d8b719412.png)
